app/controllers/calendar_controller.rb

```diff
@@ -1,11 +1,21 @@
 class CalendarController < ApplicationController
 
   def index
-    @days = ['Mon', 'Tue', 'Wed', 'Thu', 'Fri']
+    week_start = params[:date] ? Date.parse(params[:date]).at_beginning_of_week : Date.today.beginning_of_week
```

Comment:
try to define a method like:
`week_start, week_end = build_week_boundries(....)`